Source/WebKit/UIProcess/ios/AppKitSoftLink.h

```diff
@@ -25,11 +25,15 @@
 
 #pragma once
 
-#if PLATFORM(MACCATALYST)
-
 #import <wtf/SoftLinking.h>
 
 SOFT_LINK_FRAMEWORK_FOR_HEADER(WebKit, AppKit)
+
+#if PLATFORM(MACCATALYST)
 SOFT_LINK_CLASS_FOR_HEADER(WebKit, NSAccessibilityRemoteUIElement)
+#endif
 
+#if HAVE(NSTEXTPLACEHOLDER)
+SOFT_LINK_CLASS_FOR_HEADER(WebKit, NSTextPlaceholder)
+SOFT_LINK_CLASS_FOR_HEADER(WebKit, NSTextSelectionRect)
 #endif
```

Comment:
I think soft linking is unnecessary now that we don't actually allocate a real `NSTextSelectionRect`.

And it looks like `NSTextPlaceholder` is already in the SDK. The soft linking is clearly not needed for that, since you're able to `-[WKTextPlaceholder alloc]` above.